### PR TITLE
sensor: bq274xx: Support chips other than BQ27421

### PR DESCRIPTION
**Pull request by ewalt1**
This adds support for the following:
    BQ27411
    BQ27425
    BQ27426
    BQ27441

It also fixes a few bugs and improves robustness of the driver.

Signed-off-by: Nick Ewalt <nicholasewalt@google.com>

### CODE REVIEW

**Comment by ewalt1** [comment]
@parthitce 
@Luka-Lopotar-Greyp 
@pdgendt 
@palchak-google 
FYI

**Comment by ewalt1** [comment]
Can anyone take a look at this?

File: dts/bindings/sensor/ti,bq274xx.yaml [hunk 1]
```diff
@@ -11,6 +11,19 @@ compatible: "ti,bq274xx"
...
+      required: true
+      description: |
+        Part number, note that not all BQ274XX devices are compatible with this
+        driver, such as the BQ27410.
+      enum:
+         - "bq27411"
+         - "bq27421"
+         - "bq27425"
+         - "bq27426"
+         - "bq27441"
```
Comment (mbolivar-nordic) [inline dts/bindings/sensor/ti,bq274xx.yaml:25]: This is not the right approach from a devicetree perspective. The right way to do this is to make a binding for each of these parts, each with its own compatible. You can share common properties using `include` in each of these bindings of a common file.

Then, to keep some common driver code, you can redefine DT_DRV_COMPAT as many times as you need to create specific devices for each part. See drivers/i2c/i2c_tca954x.c for an example of this technique.

**Review by mbolivar-nordic — CHANGES_REQUESTED** [review]

File: dts/bindings/sensor/ti,bq274xx.yaml (hunk 1 above)
Comment (ewalt1) [reply dts/bindings/sensor/ti,bq274xx.yaml]: done

File: dts/bindings/sensor/ti,bq274xx-common.yaml [hunk 2]
```diff
@@ -40,3 +38,14 @@ properties:
         specific events happening (e.g. change in State of Charge). While in
         shutdown mode it acts as an input and toggling it will make the sensor
         exit the mode.
+
+    chemistry:
+      type: string
+      required: false
+      description: |
+        Chemistry type to use for devices that support programmable chemistry.
+        Currently this is only the BQ27426.
```
Comment (mbolivar-nordic) [inline dts/bindings/sensor/ti,bq274xx-common.yaml:47]: This should be in the binding for that compatible, then, not the common file.

Comment (ewalt1) [reply dts/bindings/sensor/ti,bq274xx-common.yaml]: done

**Review by mbolivar-nordic — APPROVED** [review]
Devicetree portions LGTM, thank you! I didn't review the rest.

**Review by MaureenHelm — CHANGES_REQUESTED** [review]
Please split up the bug fixes, cleanups, and new chip support into separate commits.

**Comment by ewalt1** [comment]
Given that I had to wait a month for a review I'm not exactly keen to go back and do a major refactor.

Feel free to reject the PR and I'll keep it local.

**Comment by ewalt1** [comment]
I wouldn't expect the size of the changeset to impact the amount of time it takes to write a single comment suggesting to break it up.

**Pull request closed without merging**